### PR TITLE
performance & docs

### PR DESCRIPTION
**Pull request by bleto**
-   new documentation
 -   performance loading modules
  -  performance structure
 -   add router helper
  -  change i18n mechanism
 -   bug fix

Issue: #11

### CODE REVIEW

File: src/helpers/tools.js [hunk 1]
```diff
@@ -10,41 +10,29 @@ import recursive from 'recursive-readdir-async';
...
-            if (beforeModule) {
-                await beforeModule.call(this, options);
-            }
+function getConfigs({ modules, suffix }) {
+  return modules.map(async ({ name, path }) => {
+      const fullPath = joinPath(path, suffix);
 
-            await moduleFn.call(this, options);
+      if (existsSync(fullPath)) {
+          const moduleConfig = await import(`${fullPath}`);
```
Comment (derpdead) [inline src/helpers/tools.js:30]: fullPath is not a string?

File: src/helpers/tools.js [hunk 2]
```diff
@@ -132,3 +96,81 @@ export function flattenDeep(arr) {
...
+* @param {Object} options - Module options
+* @param {Object} options.allModules - All active modules
+* @param {Object} options.directories - Directories names
+* @param {Boolean} options.verbose - Flag to showing logs
+* @returns {Promise<Object[]>} All modules configurations
+*/
+export function getConfigsForModules({ allModules, directories, verbose }) {
+    const configDir = directories.config || DIRECTORIES.config;
+
+    return Promise.all(
```
Comment (derpdead) [inline src/helpers/tools.js]: async await - use same approach everywhere

File: src/index.js [hunk 3]
```diff
@@ -2,74 +2,47 @@
...
  * See LICENSE for license details.
  */
-import { resolve, join } from 'path';
-import { existsSync } from 'fs-extra';
 import { info } from 'consola';
-import { log } from './helpers/log';
-import { DEFAULTS } from './helpers/constants';
-import beforeAll from './utils/beforeAll';
+import {
+    getConfigsForModules,
```
Comment (derpdead) [inline src/index.js]: getModulesConfig seems more natural

File: src/index.js [hunk 4]
```diff
@@ -2,74 +2,47 @@
...
-        ...beforeAllData.messages,
-        ...initModulesMessages,
-        ...afterAllMessages,
-    };
+    info('VueMS init');
 
-    if (options.logProcess) {
-        log(messages);
-    }
+    await beforeAllModules.call(this, {
```
Comment (derpdead) [inline src/index.js]: what about naming it as beforeModules and afterModules ? Between those function calls there are called prepareModules not prepareAllModules etc.

File: src/templates/plugin.ejs [hunk 5]
```diff
@@ -32,10 +26,10 @@ export default ({ app }, inject) => {
         }
     });
 
-    inject('getExtendedComponents', type => extendComponents[type] || {});
-    inject('extendMethods', async (type, payload) => {
-        if (extendMethods[type]) {
-            const methods = await Promise.all(extendMethods[type].map(method => method(payload)));
+    inject('getExtendSlot', key => extendComponents[key] || {});
```
Comment (derpdead) [inline src/templates/plugin.ejs:29]: what about keeping same naming convention for extendComponents? extendSlots?

Comment (bleto) [reply src/templates/plugin.ejs:29]: `extendComponents` is a property that accepts keys that specify the names of the components that we want to extend, `extendComponents` will fit better.

**Review by derpdead — APPROVED** [review]

**Pull request merged**